### PR TITLE
[DW-307] Backend for location doctype

### PR DESCRIPTION
**Pull request by ngmode**
[DW-307] Back-end for location doctype

### CODE REVIEW

File: repository-data/application/src/main/resources/hcm-config/namespaces/website/address.yaml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+          /postalCode:
+            /cluster.options:
+              jcr:primaryType: frontend:pluginconfig
+            caption: Postal Code
+            field: postalCode
+            hint: ''
+            jcr:primaryType: frontend:plugin
+            plugin.class: org.hippoecm.frontend.editor.plugins.field.PropertyFieldPlugin
+            wicket.id: ${cluster.id}.field
+          /geocordinates:
```
Comment (ngta1) [inline repository-data/application/src/main/resources/hcm-config/namespaces/website/address.yaml]: Should geocordinates be part of address compound, as further down it is used a separate field?

File: repository-data/webfiles/src/main/resources/site/freemarker/common/location.ftl [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+<#-- @ftlvariable name="location" type="uk.nhs.digital.website.beans.Location" -->
+
+<#if location??>
+<div class="grid-wrapper grid-wrapper--article">
+    <div class="grid-row">
+        <div class="column column--reset">
+            <div class="article-header article-header--secondary">
+                <div class="grid-row">
+                    <div class="column column--two-thirds column--reset">
+                        <h1 data-uipath="person.firstname">${location.name}</h1>
```
Comment (ngta1) [inline repository-data/webfiles/src/main/resources/site/freemarker/common/location.ftl]: reference to person.firstname?  if UI for location doc type to come in separate ticket, then could remove this file altogether.

File: repository-data/application/src/main/resources/hcm-config/configuration/frontend/cms-validators.yaml [hunk 3]
```diff
@@ -5,10 +5,18 @@ definitions:
       jcr:primaryType: frontend:plugin
       plugin.class: org.hippoecm.frontend.editor.validator.plugins.RegExCmsValidator
       regex_pattern: ^([a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6})?$
+    /hippo:configuration/hippo:frontend/cms/cms-validators/locationEmailAddress:
```
[the remark this answers is not in the document]
Comment (ngmode) [reply repository-data/application/src/main/resources/hcm-config/configuration/frontend/cms-validators.yaml]: done

File: repository-data/application/src/main/resources/hcm-config/configuration/frontend/cms-validators.yaml [hunk 4]
```diff
@@ -34,6 +42,10 @@ definitions:
       jcr:primaryType: frontend:plugin
       max_length: 250
       plugin.class: uk.nhs.digital.ps.TextLengthValidator
+    /hippo:configuration/hippo:frontend/cms/cms-validators/telephone:
```
[the remark this answers is not in the document]
Comment (ngmode) [reply repository-data/application/src/main/resources/hcm-config/configuration/frontend/cms-validators.yaml]: done

**Pull request merged**